src/index.js

```diff
@@ -2,13 +2,29 @@
 
 const coWrapper = require('./lib/coWrapper')
 const processImage = require('./lib/processImage')
+const loaderUtils = require('loader-utils')
 
 const loader = coWrapper(function* (content) {
-  if (typeof this.query === 'string') {
-    throw new Error('does not support inline querystring as options, define your options in webpack.config.js instead')
-  }
+  const globalOptions = loaderUtils.getOptions(this);
+  const resourceOptions = loaderUtils.parseQuery(this.resourceQuery || "?")
+  let presetOptionsList = [];
```

Comment:
maybe we can do this to avoid duplicated code
```js
const presets = (Array.isArray(resourceOptions.preset) ? resourceOptions.preset : [resourceOptions.preset]).filter((x) => x)
const presetOptionsList = presets.map((key) => (globalOptions.presets[key]))
```